apps/web/src/components/pure/workspace-slider-bar/collections/collections-list.tsx

```diff
@@ -0,0 +1,274 @@
+import {
+  EditCollectionModel,
+  useAllPageSetting,
+  useSavedCollections,
+} from '@affine/component/page-list';
+import type { Collection } from '@affine/env/filter';
+import type { GetPageInfoById } from '@affine/env/page-info';
+import {
+  DeleteIcon,
+  FilterIcon,
+  MoreHorizontalIcon,
+  UnpinIcon,
+  ViewLayersIcon,
+} from '@blocksuite/icons';
+import type { PageMeta } from '@blocksuite/store';
+import type { DragEndEvent } from '@dnd-kit/core';
+import { useDroppable } from '@dnd-kit/core';
+import * as Collapsible from '@radix-ui/react-collapsible';
+import { useBlockSuitePageMeta } from '@toeverything/hooks/use-block-suite-page-meta';
+import { useRouter } from 'next/router';
+import type { ReactElement } from 'react';
+import React, { useCallback, useMemo, useState } from 'react';
+
+import { useGetPageInfoById } from '../../../../hooks/use-get-page-info';
+import type { AllWorkspace } from '../../../../shared';
+import { filterPage } from '../../../../utils/filter';
+import type { CollectionsListProps } from '../index';
+import { Page } from './page';
+import * as styles from './styles.css';
+
+const Collections_DROP_AREA_PREFIX = 'collections-';
+const isCollectionsDropArea = (id?: string | number) => {
+  return typeof id === 'string' && id.startsWith(Collections_DROP_AREA_PREFIX);
+};
+export const processCollectionsDrag = (e: DragEndEvent) => {
+  if (
+    isCollectionsDropArea(e.over?.id) &&
+    String(e.active.id).startsWith('page-list-item-')
+  ) {
+    e.over?.data.current?.addToCollection?.(e.active.data.current?.pageId);
+  }
+};
+const CollectionOperations = ({
+  view,
+  showUpdateCollection,
+  setting,
+}: {
+  view: Collection;
+  showUpdateCollection: () => void;
+  setting: ReturnType<typeof useAllPageSetting>;
+}) => {
+  const actions = useMemo<
+    Array<
+      | {
+          icon: ReactElement;
+          name: string;
+          click: () => void;
+          className?: string;
+          render?: undefined;
+        }
+      | {
+          render: ReactElement;
+        }
+    >
+  >(
+    () => [
+      {
+        icon: <FilterIcon />,
+        name: 'Edit Filter',
+        click: showUpdateCollection,
+      },
+      {
+        icon: <UnpinIcon />,
+        name: 'Unpin',
+        click: () => {
+          return setting.updateCollection({
+            ...view,
+            pinned: false,
+          });
+        },
+      },
+      {
+        render: <div key="divider" className={styles.menuDividerStyle}></div>,
+      },
+      {
+        icon: <DeleteIcon style={{ color: 'var(--affine-warning-color)' }} />,
+        name: 'Delete',
+        click: () => {
+          return setting.deleteCollection(view.id);
+        },
+        className: styles.deleteFolder,
+      },
+    ],
+    [setting, showUpdateCollection, view]
+  );
+  return (
+    <div style={{ minWidth: 150 }}>
+      {actions.map(action => {
+        if (action.render) {
+          return action.render;
```

Comment:
change the name `render`, I want to call it if I see it. I'll suggest changing it into `element`